src/adminEntity.ts

```diff
@@ -65,6 +66,26 @@ abstract class AdminEntity {
 
     return [...widgets, ...manyToManyWidgets]
   }
+
+  validate() {
+    if (this.searchFields) {
+      this.searchFields.forEach(field => {
+        if (!this.metadata.columns.map(column => column.propertyName).includes(field)) {
+          throw new InvalidSearchFieldsException(
+            `Property ${field} does not exist on ${this.entity.name}.`,
+          )
+        } else {
+          // We do not support searching relations.
+          const relation = this.metadata.findRelationWithPropertyPath(field)
+          if (relation) {
+            throw new InvalidSearchFieldsException(
+              `Property ${field} on ${this.entity.name} is a relation, which is not supported for searching.`,
```

Comment:
Are you sure `this.entity.name` is a thing? Probably `this.name` would make more sense?